src/Client.php

```diff
@@ -684,8 +676,31 @@ protected function deserialize(string $className, string $data)
      */
     private function fixEncounters(string $uri): array
     {
-        $fixedUri = str_replace('/api/v2/', '', $uri);
+        $callback = $this->callback;
+
+        return json_decode($callback($uri), true);
```

Comment:
I'd go one step further, and put `json_decode()` in the callbable that  `getDefaultCallback()` returns. That way, we aren't bound to a single implementation, and the given callable could process it's data itself, rather than returning a JSON string. What do you think about it ?